jsonschema/_validators.py

```diff
@@ -514,6 +516,22 @@ def unevaluatedItems(validator, unevaluatedItems, instance, schema):
                 yield error
 
 
+def unevaluatedProperties(validator, unevaluatedProperties, instance, schema):
+    if not validator.is_type(instance, "object"):
+        return
+
+    evaluated_property_keys = find_evaluated_property_keys_by_schema(validator, instance, schema)
+    for property, subschema in instance.items():
+        if property not in evaluated_property_keys:
+            for error in validator.descend(
+                    instance[property],
+                    unevaluatedProperties,
+                    path=property,
+                    schema_path=property,
+            ):
+                yield error
```

Comment:
With 'additionalProperties: false', you get a message saying what unexpected properties were found. With 'unevaluatedProperties: false', you get 'false schema does not allow: <value of unexpected property>'. It would be nice to follow the 'additionalProperties' code here if possible.

I didn't check, but I'd guess the same thing applies to unevaluatedItems WRT additionalItems.

Reply:
I made an update for this in https://github.com/Julian/jsonschema/pull/817/commits/9200165c40f463e0b7b7fa1e5fba5ef5eacdd33e